### PR TITLE
Adding type validation logic to object operations

### PR DESCRIPTION
**Pull request by TalZaccai**
- [x] List commands
- [x] Set commands
- [x] SortedSet commands
- [x] Hash commands
- [x] Geo commands
- [x] Shared commands
- [x] Custom object commands

### CODE REVIEW

File: libs/server/Storage/Session/ObjectStore/SetOps.cs [hunk 1]
```diff
@@ -398,19 +391,57 @@ internal unsafe GarnetStatus SetMove(ArgSlice sourceKey, ArgSlice destinationKey
...
                 }
+                else
+                {
+                    dstSetObject = new SetObject();
+                }
+
+                var arrMember = member.ToArray();
+
+                var removed = srcSetObject.Set.Remove(arrMember);
+                if (!removed) return GarnetStatus.OK;
```
Comment (TedHartMS) [inline libs/server/Storage/Session/ObjectStore/SetOps.cs:428]: Do these *Move operations need a "bool removeSrcKey" if they go empty?

**Review by TedHartMS — APPROVED** [review]

File: libs/server/Resp/Objects/SortedSetCommands.cs [hunk 2]
```diff
@@ -289,11 +315,9 @@ private unsafe bool SortedSetRange<TGarnetApi>(RespCommand command, int count, b
                     //command partially executed
                     return false;
                 }
-                else
-                {
-                    while (!RespWriteUtils.WriteNull(ref dcurr, dend))
-                        SendAndReset();
-                }
+
+                while (!RespWriteUtils.WriteNull(ref dcurr, dend))
```
Comment (yrajas) [inline libs/server/Resp/Objects/SortedSetCommands.cs:319]: >while (!RespWriteUtils.WriteNull(ref dcurr, dend)) [](http://example.com/codeflow?start=16&length=50)

Shouldn't it be wrong # of args?

Comment (TalZaccai) [reply libs/server/Resp/Objects/SortedSetCommands.cs:319]: Good catch!

File: libs/server/Storage/Session/ObjectStore/SetOps.cs (hunk 1 above)
Comment (TalZaccai) [reply libs/server/Storage/Session/ObjectStore/SetOps.cs:428]: Will resolve this an other similar scenarios in a separate PR

**Pull request merged**